src/audio/copier/copier.c

```diff
@@ -1553,6 +1553,22 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 							  NOTIFIER_ID_DMA_COPY, copier_dma_cb, 0);
 
 				cd->hd->process = cd->converter[IPC4_COPIER_GATEWAY_PIN];
+			} else if (dev->ipc_config.type == SOF_COMP_DAI) {
```

Comment:
Needs some inline commenst to explain what this is doing.

Reply:
added